### ISSUE TITLE
Bug: `Jump to Link` does not work in Source Mode

### ISSUE THREAD

**Issue by chrisgrieser** [mrjackphil/obsidian-jump-to-link#37]
this seems to be a very similar case to #31. While the issue has been resolved at that time, it re-occured. I deleted my `app.json` like a commenter described in #31, and jump to link works again. 

After an hour or so of trying to reproduce what exactly causes the issue, I found what exactly causes the bug: 

When in the Obsidian settings, under `Editor → General → Default Editing Mode`, I select `Source Mode` (while *not* using the Legacy Editor), and I *then* open a new panel, `Jump to Link` won't work in Source Mode anymore – but it still works in Reading Mode and Live Preview. When I set the default editor setting to `Live Preview`, open a new pane, and then switch to Source Mode via Command Palette, `Jump to Link` does work in Source Mode. 

So all in all, it seems, it is not Source Mode which causes the problem, but rather initializing a new pane in Source mode. Whether a pane which was *not* initialized in Source mode is later switched to Source Mode seems to be irrelevant for this bug.

---

My current workaround is to use the [Force View Mode Plugin](https://obsidian.md/plugins?id=obsidian-view-mode-by-frontmatter), which forces source mode even though Live Preview is my default editor mode, because it hasn't been updated since the introduction of live preview. So basically, my workaround only works by abusing the bug of an outdated other plugin. (lol)

---

This is the log I get when the command fails to work in Source Mode:

```
Command failed to execute:  mrj-jump-to-link:activate-jump-to-link
app.js:1 TypeError: cmEditor.coordsChar is not a function
    at getVisibleLineText (eval at <anonymous> (app.js:1:1494667), <anonymous>:47:37)
    at SourceLinkProcessor.getSourceLinkHints (eval at <anonymous> (app.js:1:1494667), <anonymous>:308:41)
    at SourceLinkProcessor.init (eval at <anonymous> (app.js:1:1494667), <anonymous>:316:32)
    at JumpToLink.handleJumpToLink (eval at <anonymous> (app.js:1:1494667), <anonymous>:430:88)
    at JumpToLink.action (eval at <anonymous> (app.js:1:1494667), <anonymous>:568:22)
    at Mz (app.js:1:1618036)
    at e.executeCommandById (app.js:1:1619289)
    at eval (eval at <anonymous> (app.js:1:1494667), <anonymous>:74:11954)
    at e.o [as executeCommandById] (eval at <anonymous> (app.js:1:1494667), <anonymous>:23:77227)
    at e.onTrigger (app.js:1:1158645)
```

**Comment by mrjackphil** [comment]
Thanks for such detailed report. I see what causing the error. 
So it should be fixed next time I'll put my hands on plugin update.

**Comment by mrjackphil** [comment]
Should be fixed in 0.3.3

**Comment by chrisgrieser** [comment]
thank you, works now as intended 🙂
